src/core/unittest/SettingsTest.cpp

```diff
@@ -417,3 +417,42 @@ TEST(SettingsTest, GlobalSettingsSizesSet)
                 &InternalSettings));
     }
 }
+
+TEST(SettingsTest, GlobalRawDataPathProcsSetAndGet)
+{
+    uint32_t SetCpus[] = {1, 2, 3, 4, 5};
+    ASSERT_EQ(
+        QUIC_STATUS_SUCCESS,
+        QuicLibrarySetGlobalParam(
+            QUIC_PARAM_GLOBAL_RAW_DATAPATH_PROCS,
+            sizeof(SetCpus),
+            SetCpus));
+    uint32_t BufferLength = 0;
+    ASSERT_EQ(
+        QUIC_STATUS_BUFFER_TOO_SMALL,
+        QuicLibraryGetGlobalParam(
+            QUIC_PARAM_GLOBAL_RAW_DATAPATH_PROCS,
+            &BufferLength,
+            NULL));
+    ASSERT_EQ(ARRAYSIZE(SetCpus), BufferLength / sizeof(uint32_t));
+    uint32_t GetCpus[ARRAYSIZE(SetCpus)] = {0};
+    ASSERT_EQ(
+        QUIC_STATUS_SUCCESS,
+        QuicLibraryGetGlobalParam(
+            QUIC_PARAM_GLOBAL_RAW_DATAPATH_PROCS,
+            &BufferLength,
+            GetCpus));
+    ASSERT_EQ(0, memcmp(GetCpus, SetCpus, sizeof(SetCpus)));
+}
+
+TEST(SettingsTest, GlobalRawDataPathProcsSetAfterDataPathInit)
+{
+    uint32_t SetCpus[] = {1, 2, 3, 4, 5};
+    MsQuicLib.Datapath = (CXPLAT_DATAPATH*)1; // Pretend datapath has been initialized
```

Comment:
Does "MsQuicLib" get reused for other testcases? Because I see you aren't setting MsQuicLib.Datapath back to NULL.

In general this is a little too hacky IMO- you are relying on the internal code not dereferencing this pointer, which could easily change in the future.

Reply:
These are unit tests all do this for the most part it's fine.

Reply:
For these unit tests, I think, in each test case, if we read values from global variables, we should first memset them to zero in the very beginning.

Reply:
If gtests have a nice way to do per test case level clean-up, I can do it.

Reply:
Oh, I can zero msquiclib in test case teardown.